host/sgx/calls.c

```diff
@@ -545,6 +545,7 @@ static void* _assign_tcs(oe_enclave_t* enclave)
                     binding->flags |= _OE_THREAD_BUSY;
                     binding->thread = thread;
                     binding->count = 1;
+                    binding->enclave = enclave;
```

Comment:
This works, but I don't think this should be a mutable field on binding; there's a tight coupling of TCS to the enclave associated with that TCS.  

It would be ideal if the TCS and its reverse mapping of the TCS to the host representation of the `oe_enclave_t` are defined together in the binding and immutable after enclave creation.

Reply:
Good point. Done.